recipes/google-guetzli/all/conanfile.py

```diff
@@ -1,54 +1,68 @@
 import os
 
 
-class GoogleguetzliConan(ConanFile):
+class GoogleGuetzliConan(ConanFile):
     name = "google-guetzli"
     license = "Apache-2.0"
     url = "https://github.com/conan-io/conan-center-index"
     homepage = "https://opensource.google/projects/guetzli"
     description = "Perceptual JPEG encoder"
     topics = "jpeg","compression"
+    exports_sources = "patches/**"
     settings = "os", "compiler", "build_type", "arch"
     generators = "pkg_config"
     requires = ["libpng/1.6.37"]
 
-    _source_subfolder = "source_subfolder"
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
 
     @property
-    def on_windows(self):
-        return self.settings.os == "Windows"
+    def _is_msvc(self):
+        return self.settings.compiler == "Visual Studio"
 
     def configure(self):
-        if self.settings.os not in ["Linux","Windows"]:
-            raise ConanInvalidConfiguration("conan recipe for  google-guetzli v{0} is not \
+        if self.settings.os not in ["Linux", "Windows"]:
+            raise ConanInvalidConfiguration("conan recipe for google-guetzli v{0} is not \
                 available in {1}.".format(self.version, self.settings.os))
 
     def source(self):
         tools.get(**self.conan_data["sources"][self.version])
         extracted_dir = "guetzli-" + self.version
         os.rename(extracted_dir, self._source_subfolder)
 
+    def _patch_sources(self):
+        for patch in self.conan_data["patches"][self.version]:
+            tools.patch(**patch)
+
     def build(self):
-        if self.on_windows:
-            from conans import MSBuild
+        self._patch_sources()
+        if self._is_msvc:
             msbuild = MSBuild(self)
             with tools.chdir(self._source_subfolder):
                 msbuild.build("guetzli.sln")
         else:
+            autotools = AutoToolsBuildEnvironment(self)
             with tools.chdir(self._source_subfolder):
-                print("cwd = {}".format(os.getcwd()))
-                with tools.environment_append({"PKG_CONFIG_PATH": self.build_folder}):
-                    self.run("make")
+                env_vars = {"PKG_CONFIG_PATH": self.build_folder}
+                env_vars.update(autotools.vars)
+                with tools.environment_append(env_vars):
+                    make_args = [
+                        "config={}".format("debug" if self.settings.build_type == "Debug" else "release"),
+                        "verbose=1',"
+                    ]
+                    autotools.make(args=make_args)
             
     def package(self):
-        if self.on_windows:
-            self.copy("{}/bin/{}/Release/guetzli.exe".format(self._source_subfolder, self.settings.arch),
+        subdir = "Debug" if self.settings.build_type == "Debug" else "Release"
+        if self._is_msvc:
+            self.copy("{}/bin/{}/Release/{}.exe".format(self._source_subfolder, self.settings.arch, subdir),
```

Comment:
```suggestion
            self.copy("{}/bin/{}/{}/guetzli.exe".format(self._source_subfolder, self.settings.arch, subdir),
```
I was wrong here.
I didn't test on Windows.